criu/arch/mips/restorer.c

```diff
@@ -15,3 +15,35 @@ int restore_nonsigframe_gpregs(UserMipsRegsEntry *r)
 {
     return 0;
 }
+
+#define SHMLBA 0x40000
+unsigned long arch_shmat(int shmid, void *shmaddr,
+			int shmflg, unsigned long size)
+{
+	unsigned long smap;
+
+	/* SHMLBA-aligned, direct call shmat() */
+	if (!((unsigned long)shmaddr & (SHMLBA - 1)))
+		return sys_shmat(shmid, shmaddr, shmflg);
+
+	smap = sys_shmat(shmid, NULL, shmflg);
+	if (IS_ERR_VALUE(smap)) {
+		pr_err("shmat() with NULL shmaddr failed: %d\n", (int)smap);
+		return smap;
+	}
+
+	/* We're lucky! */
+	if (smap == (unsigned long)shmaddr)
+		return smap;
+
+	 /* Warn ALOUD */
+        pr_warn("Restoring shmem %p unaligned to SHMLBA.\n", shmaddr);
+        pr_warn("Make sure that you don't migrate shmem from non-VIPT cached CPU to VIPT cached \n");
+        pr_warn("Otherwise YOU HAVE A CHANCE OF DATA CORRUPTIONS in writeable shmem\n");
```

Comment:
I don't want to be any picky, but the indention-style differs from the rest of the file (those should be tabs here).
Other than this, LGTM

Reply:
I'd appreciate it if you have any tools for check indention-style to recommend me.

Reply:
Well, I use `vim` to auto-indent code.
You can use `scripts/checkpatch.pl` from the Linux kernel repository - it will tell you if there are any missing spaces between brackets or anything alike.
In CRIU we stick to kernel code style.